src/dex-ui/store/swapSlice/swapSlice.ts

```diff
@@ -43,6 +45,16 @@ const createSwapSlice: SwapSlice = (set, get): SwapStore => {
         SwapActionType.SET_PRECISION
       );
     },
+    fetchTokenPairs: async () => {
+      const pairs = await HederaService.getTokenPairs();
+      set(
+        ({ swap }) => {
+          swap.tokenPairs = pairs;
+        },
+        false,
+        SwapActionType.SET_TOKEN_PAIRS
```

Comment:
This is ok for now. In a future iteration, we will need to utilize multiple action types to track when this call is successful or returns an error. For example, could use the following action types:

```
FETCH_TOKEN_PAIRS_STARTED = "swap/FETCH_TOKEN_PAIRS_STARTED"
FETCH_TOKEN_PAIRS_SUCCEEDED = "swap/FETCH_TOKEN_PAIRS_SUCCEEDED"
FETCH_TOKEN_PAIRS_FAILED=  "swap/FETCH_TOKEN_PAIRS_FAILED"
```

Take a look at the patterns used in other store functions to see how this is implemented.

Reply:
Done